crypto/x509/x509_vfy.c

```diff
@@ -525,6 +526,14 @@ static int check_chain_extensions(X509_STORE_CTX *ctx)
                 ret = 1;
             break;
         }
+        if ((ctx->param->flags & X509_V_FLAG_X509_STRICT) && num > 1) {
```

Comment:
So there isn't an opt-in other than making the `X509_V_FLAG_X509_STRICT` flag more strict?

I am definitely not familiar with X.509 code and its usage by external applications to judge if this is considered a breaking change or not.

@t8m who could we ping about an answer to this?

Reply:
Ping @vdukhovni

Reply:
@kroeckx also said that he is not seeing such certificates in his huge collected set of certificates. Besides the `num > 1` skips the check for self-signed certs.

Reply:
Note that the documentation of `X509_V_FLAG_X509_STRICT` is very vague:
```
The B<X509_V_FLAG_X509_STRICT> flag disables workarounds for some broken
certificates and makes the verification strictly apply B<X509> rules.
```
but any user should assume that *all* requirements of RFC 5280 etc. are checked
(while in fact so far we only check a rather limited subset).

Reply:
I presume that only rather few applications actually use this flag, and even if so, 
the likelihood that due to this new check anything breaks for them is extremely close to zero.

Reply:
> I presume that only rather few applications actually use this flag, and even if so,
> the likelihood that due to this new check anything breaks for them is extremely close to zero.

This would have broken Akamai in production, so your presumption and belief are wrong.

Reply:
@richsalz, do you really mean that Akamai uses both `X509_V_FLAG_X509_STRICT` and certs with explicit elliptic curve parameters?
I cannot imagine that.

Reply:
@ddvo: yes, that is *exactly* what I meant.  I guess our customer list is bigger than your imagination :)

Reply:
Interesting that Akamai does use certs with explicit EC curve parameters 
while at the same time being strict about RFC 5280 requirements and thus forbidding them.
To me this makes sense only as a unit test case, as done in OpenSSL itself.

Reply:
We have many thousands of customers and we are not always in charge of the certificates.

>To me this makes most sense only as a unit test case,

For the third time, this is wrong. It would have broken us *in production.*

Reply:
> It would have broken us _in production._

I'm still a bit puzzled about the way you stated this (even twice using these very words).
Shall this mean that 
* you cannot use `X509_V_FLAG_X509_STRICT` anymore due to its strengthened checks on explicit EC curve parameters,

 or
* that the bug introduced as an inadvertent side-effect would have affected you hard - 
which would be an entirely different thing because that pertains to 
(wrongly) accepting non-CA intermediate certs that do not have explicit EC curve parameters?

Reply:
The bug would have broken us and, in fact, that's how we found the problem: we weren't getting the right error.

Reply:
So as I had already suspected right from the beginning,
your reply

> This would have broken Akamai in production, so your presumption and belief are wrong.

on my comment

>> I presume that only rather few applications actually use this flag, and even if so,
the likelihood that due to this new check anything breaks for them is extremely close to zero.

was entirely misleading because it connected two quite different things
* being affected by the bug
* my presumption on the effect of the new check, 
which should have been taken independently of the bug.

Reply:
I'm really done with this thread.  My latest reply was wrong.  We found the bug because we had a testcase and we would have had to make adjustments.

If you want to persist that you were right, that the additional tests didn't hurt anything, I can't stop you.